hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/disk/OnDiskKey.java

```diff
@@ -100,16 +100,19 @@ public void serialize(@NonNull final ByteBuffer byteBuffer) throws IOException {
         output.skip(4);
         codec.write(key, output);
         final var pos = output.position();
-        output.position(0);
         output.writeInt((int) pos - 4);
         output.position(pos);
     }
```

Comment:
I don't understand how this code works... Several issues about it:

* When the method is called, `byteBuffer`'s position isn't guaranteed to be 0. It can be any value
* It means, `pos - 4`, which is used as the length, may be incorrect
* Then, `output.writeInt()` writes an int at the current position, which is _after_ what is written by the codec. The int is not written to the first (skipped) 4 bytes
* It means, here the length is written after key bytes, while `deserialize()` reads the length before the bytes

Anyway. This code is going to be rewritten to use MerkleDb APIs shortly as a part of https://github.com/hashgraph/hedera-services/pull/7968, and the methods will be fixed.